vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts

```diff
@@ -317,14 +330,25 @@ class ChannelController implements ITelemetryPlugin {
     initialize(config: IConfiguration, core: IAppInsightsCore, extensions: IPlugin[]) {
         this.channelQueue = new Array<IChannelControls[]>();
         if (config.channels) {
+            let invalidChannelIdentifier = undefined;
             config.channels.forEach(queue => {
+
                 if(queue && queue.length > 0) {
                     queue = queue.sort((a,b) => { // sort based on priority within each queue
                         return a.priority - b.priority;
                     });
 
                     // Initialize each plugin
-                    queue.forEach(queueItem => queueItem.initialize(config, core, extensions));
+                    queue.forEach(queueItem => {
+                        if (queueItem.priority < ChannelControllerPriority) {
+                            invalidChannelIdentifier = queueItem.identifier;
+                        }
+                        queueItem.initialize(config, core, extensions)
+                    });
+
+                    if (invalidChannelIdentifier) {
```

Comment:
How this is related to having duplicated events?

Reply:
The duplicated events was coming from not adding the channel side loaded correctly. We've seen this in the past, when channel is not added correctly in core initialization, we saw duplicate events. I updated fabrikam code to reflect usage there as well.